hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/transaction/BlockItem.java

```diff
@@ -20,14 +20,54 @@
 import com.hedera.hapi.block.stream.output.protoc.TransactionOutput;
 import com.hedera.hapi.block.stream.output.protoc.TransactionResult;
 import com.hedera.mirror.common.domain.StreamItem;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
 import com.hederahashgraph.api.proto.java.Transaction;
 import java.util.List;
 import lombok.Builder;
 
-@Builder(toBuilder = true)
+@Builder(buildMethodName = "builderInternal", toBuilder = true)
 public record BlockItem(
         Transaction transaction,
         TransactionResult transactionResult,
         List<TransactionOutput> transactionOutput,
-        List<StateChanges> stateChanges)
-        implements StreamItem {}
+        List<StateChanges> stateChanges,
+        BlockItem parent,
+        BlockItem previous,
+        boolean isSuccessful)
+        implements StreamItem {
+
+    // Custom Builder
+    public static class BlockItemBuilder {
+        public BlockItem build() {
+            parent = parseParent();
+            isSuccessful = parseSuccess();
+            return builderInternal();
+        }
```

Comment:
Why can't we use the record custom constructor to do this logic? Then it doesn't force us to only create via the builder.
```suggestion
    public BlockItem {
        parent = parseParent();
        isSuccessful = parseSuccess();
    }
```

Reply:
Just to note, I've updated my PR with a similar change in this overlapping area. Not sure if it is exactly what we want, but this is how I've implemented it thus far:
 https://github.com/hashgraph/hedera-mirror-node/pull/10254

Reply:
I deleted the BlockItemBuilder and used the default constructor. The builder pattern uses the default constructor as well so there shouldn't be an issue.